setup.sh

```diff
@@ -33,8 +33,11 @@ else
             if [ -z "$ASSET_URLS" ]; then
                 ASSET_URLS=$(curl -s "$ASSETS_URL" | grep -o -E 'https://github.com/[^"]+\.so')
             fi
-        else
-            ASSET_URLS=$(curl -s "$ASSETS_URL" | grep -o -E 'https://github.com/[^"]+\.so')
+		else
+			ASSET_URLS=$(curl -s "$ASSETS_URL" | grep -o -E "https://github.com/[^"]+$ARCH[^"]*\.so")
```

Comment:
I made a mistake here, and have corrected in my earlier comment. Additionally, the same logic applies to both ARM64 and AMD64 architectures, so the code starting from line 31 can now be removed.